analyses/mwi/dryspells_trigger/15_mwi_obs_trigger_monitoring.md

```diff
@@ -0,0 +1,154 @@
+    country_iso3 = "mwi",
+    monitoring_start = MONITORING_START,
+    monitoring_end = MONITORING_END,
+    range_x = RANGE_X,
+    range_y = RANGE_Y
+)
+
+arc2_centr.download_data()
+
+# touching method
+arc2_touch = DrySpells(
+    country_iso3 = "mwi",
+    monitoring_start = MONITORING_START,
+    monitoring_end = MONITORING_END,
+    range_x = RANGE_X,
+    range_y = RANGE_Y,
+    agg_method = "touching"
+)
+
+# approximate mask
+arc2_approx = DrySpells(
+    country_iso3 = "mwi",
+    monitoring_start = MONITORING_START,
+    monitoring_end = MONITORING_END,
+    range_x = RANGE_X,
+    range_y = RANGE_Y,
+    agg_method = "approximate_mask"
+)
+```
+
+Now with each of these, we can just re-process our data and calculate rolling sums and dry spells.
+
+```python
+arc2_centr.aggregate_data(POLY_PATH, "ADM2_PCODE")
+arc2_centr.calculate_rolling_sum()
+arc2_centr.identify_dry_spells()
+
+arc2_touch.aggregate_data(POLY_PATH, "ADM2_PCODE")
+arc2_touch.calculate_rolling_sum()
+arc2_touch.identify_dry_spells()
+
+arc2_approx.aggregate_data(POLY_PATH, "ADM2_PCODE")
+arc2_approx.calculate_rolling_sum()
+arc2_approx.identify_dry_spells()
+```
+
+How many dry spells have we detected thus far in our monitoring period?
+
+```python
+print(
+    f"Centroid method: {arc2_centr.count_dry_spells()}\n"
+    f"Touching method: {arc2_touch.count_dry_spells()}\n"
+    f"Approx m method: {arc2_approx.count_dry_spells()}"
+)
+```
+
+And are we getting close to triggering? This is useful during monitoring. Let's check the number of areas that have gotten 12 days or more cumulative sum below 2mm rainfall.
+
+```python
+DAYS = 12
+
+print(
+    f"Centroid method: {arc2_centr.count_days_under_threshold(DAYS)}\n"
+    f"Touching method: {arc2_touch.count_days_under_threshold(DAYS)}\n"
+    f"Approx m method: {arc2_approx.count_days_under_threshold(DAYS)}"
+)
+```
+
+So, if we have triggered, we also want to plot the cumulative rainfall across our area of interest and monitoring period, as well as numbers of days without rain above 2mm. We can plot with any of the methods of aggregation, but chose to use centroid for below.
+
+```python
+# get number of consecutive days under threshold per area
+da_days = arc2_centr.days_under_threshold()
+f, ax = plt.subplots()
+da_days = da_days.rio.clip(gdf_adm3.geometry)
+da_days = da_days.where(da_days.values >= 0, np.NaN)
+divnorm = colors.TwoSlopeNorm(vmin=0, vcenter=14, vmax = 28)
+da_days.plot(ax = ax,
+             cmap='Greys',
+             norm = divnorm)
+gdf_adm3.plot(ax=ax, facecolor="none", alpha=0.5)
+
+plt.title("Most consecutive days under 2mm cumulative rainfall")
+```
+
+```python
+da_cum = arc2_centr.cumulative_rainfall()
+f, ax = plt.subplots()
+da_cum = da_cum.rio.clip(gdf_adm3.geometry)
+da_cum = da_cum.where(da_cum.values >= 0, np.NaN)
+da_cum.plot(ax = ax,
+            cmap='Greys')
+gdf_adm3.plot(ax=ax, facecolor="none", alpha=0.5)
+plt.title(f"Cumulative rainfall from {arc2_centr.monitoring_start}")
```

Comment:
Here you are indicating it is the cumulative rainfall from `monitoring_start`. But looking in the code of `cumulative_rainfall()` it is the cum rainfall from `date_min` where `date_min=monitoring_start-rolling_window`. So should at least be aligned. I think it would make most sense to set the start date of `cumulative_rainfall()` to `monitoring_start` instead of `date_min`

Reply:
Very good catch on the mislabeling, missed it in the last effort code up prior to leave! Here, I think we should still look from `date_min`, because that is when we start tracking for dry spells (that's the first day included in a cumulative sum).